frontend/src/components/MeetingRoom/AudioInputOutputDevices/OutputDevices/OutputDevices.tsx

```diff
@@ -29,37 +28,27 @@ const OutputDevices = ({
   handleToggle,
   customLightBlueColor,
 }: OutputDevicesProps): ReactElement => {
-  const { audioOutput, setAudioOutput } = useAudioOutputContext();
-  const [devicesAvailable, setDevicesAvailable] = useState<AudioOutputDevice[] | null>(null);
-  const { allMediaDevices } = useDevices();
-
-  useEffect(() => {
-    setDevicesAvailable(allMediaDevices.audioOutputDevices);
-  }, [allMediaDevices]);
+  const { currentAudioOutputDevice, setAudioOutputDevice } = useAudioOutputContext();
+  const {
+    allMediaDevices: { audioOutputDevices },
+  } = useDevices();
 
-  const changeAudioOutput = async (deviceId: string) => {
-    await setAudioOutputDevice(deviceId);
-    setAudioOutput(deviceId);
-  };
+  const isAudioOutputSupported = isGetActiveAudioOutputDeviceSupported();
 
-  useEffect(() => {
-    const getActiveAudioOutputDeviceLabel = async () => {
-      const activeOutputDevice = await getActiveAudioOutputDevice();
-      setAudioOutput(activeOutputDevice.deviceId);
-    };
-    getActiveAudioOutputDeviceLabel();
-  }, [setAudioOutput]);
+  const availableDevices = isAudioOutputSupported ? audioOutputDevices : defaultOutputDevices;
 
-  const handleChangeAudioOutput = (event: MouseEvent<HTMLLIElement>) => {
+  const handleChangeAudioOutput = async (event: MouseEvent<HTMLLIElement>) => {
```

Comment:
Same as the above file, a lot of this was not needed, made testing hard, and caused multiple render chains before the state was right, when the state could be derived from props/hooks directly.